src/libraries/System.Collections/src/System/Collections/Generic/Stack.cs

```diff
@@ -288,6 +288,45 @@ private void PushWithResize(T item)
             _size++;
         }
 
+        /// <summary>
+        /// Ensures that the capacity of this Stack is at least the specified <paramref name="capacity"/>.
+        /// If the current capacity of the Stack is less than specified <paramref name="capacity"/>,
+        /// the capacity is increased by continuously twice current capacity until it is at least the specified <paramref name="capacity"/>.
+        /// </summary>
+        /// <param name="capacity">The minimum capacity to ensure</param>
+        public int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(capacity), capacity, SR.ArgumentOutOfRange_NeedNonNegNum);
+            }
+
+            if (_array.Length < capacity)
+            {
+                int newCapacity = (_array.Length == 0) ? DefaultCapacity : 2 * _array.Length;
+                while (newCapacity < capacity)
```

Comment:
Same here, account for arithmetic overflow.

Reply:
Fixed.